modules/videoio/src/cap_v4l.cpp

```diff
@@ -881,6 +886,112 @@ bool CvCaptureCAM_V4L::tryIoctl(unsigned long ioctlCode, void *parameter) const
     return true;
 }
 
+bool CvCaptureCAM_V4L::deviceHandlePoll(const std::vector<int>& deviceHandles, std::vector<int>& state, int64_t timeout)
+{
+     if(!deviceHandles.empty())
+     {
+        const auto poll_flags = POLLIN | POLLRDNORM | POLLERR;
+
+        std::vector<pollfd> fds;
+
+        for(const auto& dhand_num : deviceHandles)
+        {
+            if(dhand_num)
+            {
+                fds.push_back(pollfd{dhand_num, poll_flags, 0});
+            }
+            else
+                return false;
+        }
+        int ret = poll(fds.data(), fds.size(), timeout);
+
+        if(ret == -1)
+        {
+            perror("poll error");
+            return false;
+        }
+
+        for (size_t struct_num = 0; struct_num < fds.size(); ++struct_num)
+        {
+            if (ret != 0)
+            {
+                if((fds[struct_num].revents & (POLLIN | POLLRDNORM)) != 0)
+                {
+                    state[struct_num] = CAP_CAM_READY;
+                }
+                else
+                    if((fds[struct_num].revents & POLLERR) != 0)
+                    {
+                        state[struct_num] = CAP_CAM_ERROR;
+                    }
+                    else
+                        state[struct_num] = CAP_CAM_NOT_READY;
+            }
+            else
+                state[struct_num] = CAP_CAM_NOT_READY;
+        }
+     }
+     return true;
+}
+
+bool CvCaptureCAM_V4L::camerasPoll(const std::vector<CvCapture*>& pointers, std::vector<int>& state, int64_t timeout)
+{
+    if(!pointers.empty())
+    {
+        std::vector<int> deviceHandles;
+        for(const auto& ptr_num : pointers)
+        {
+            if(ptr_num)
+            {
+                CV_DbgAssert(dynamic_cast< CvCaptureCAM_V4L * >(ptr_num) != nullptr);
+                CvCaptureCAM_V4L *ptr = static_cast<CvCaptureCAM_V4L * >(ptr_num);
+                deviceHandles.push_back(ptr->deviceHandle);
+                ptr->setFirstCapture();
+            }
+            else
+                return false;
+        }
+        CV_DbgAssert(dynamic_cast< CvCaptureCAM_V4L * >(pointers[0]) != nullptr);
+        CvCaptureCAM_V4L *ptr = static_cast<CvCaptureCAM_V4L * >(pointers[0]);
+        if(ptr->deviceHandlePoll(deviceHandles, state, timeout))
+                return true;
+    }
+    return false;
+}
+
+bool CvCaptureCAM_V4L::setFirstCapture()
```

Comment:
What does this method do? There is duplicated code in `grabFrame`. Duplicated code is bad practice.